athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTableUtils.java

```diff
@@ -187,6 +175,24 @@ public static Schema peekTableForSchema(String tableName, ThrottlingInvoker invo
         return schemaBuilder.build();
     }
 
+    public static SchemaBuilder buildSchemaFromItems(List<Map<String, AttributeValue>> items)
+    {
+        SchemaBuilder schemaBuilder = new SchemaBuilder();
+        Set<String> discoveredColumns = new HashSet<>();
+        for (Map<String, AttributeValue> item : items) {
+            for (Map.Entry<String, AttributeValue> column : item.entrySet()) {
+                if (!discoveredColumns.contains(column.getKey())) {
```

Comment:
can we improve it to not have this many layer?

Reply:
Yes, I'll refactor it.